### PR TITLE
2015 update owl2 printers to swrl rules

### PR DESCRIPTION
**Pull request by daniilsvc**
Issue #2015

### CODE REVIEW

**Comment by daniilsvc** [comment]
For MS printer only DLSafeRules are implemented so far.

**Comment by mcodescu** [comment]
In `OWL2.Theorem` I get warnings about non-exhaustive pattern matches, and when trying to run Hets on an ontology with SWRL rules, I get an error `hets: OWL2/Theorem.hs:(130,12)-(173,46): Non-exhaustive patterns in case`. Please solve this.

**Comment by mcodescu** [comment]
Now I get `hets: OWL2/Theorem.hs:(29,16)-(72,63): Non-exhaustive patterns in case`. Please make sure that there are no warnings at all when compiling this file (and actually all warnings should be solved before code is merged).

**Comment by mcodescu** [comment]
Now everything is fine.
There's a formatting issue with displaying rules. I have the following rule:
```
DLSafeRule(Body(ClassAtom(FOD:FoodOrDrink Variable(<urn:swrl#f>)) ObjectPropertyAtom(<http://ontologies.baall.de/Hyperontology/Diet.owl#isAllowedFODSource_1discouraged> Variable(<urn:swrl#p>) Variable(<urn:swrl#bf>)) ClassAtom(<http://ontologies.baall.de/Hyperontology/DUL.owl#NaturalPerson> Variable(<urn:swrl#p>)) ObjectPropertyAtom(<http://ontologies.baall.de/Hyperontology/Object.owl#comprisesConstituent_2MediumHigh_or_greater_Intensity> Variable(<urn:swrl#f>) Variable(<urn:swrl#bf>)))Head(ObjectPropertyAtom(<http://ontologies.baall.de/Hyperontology/Diet.owl#isAllowedFOD_1discouraged> Variable(<urn:swrl#p>) Variable(<urn:swrl#f>))))
```
and it is displayed as 
```
          DLSafeRule( Body( ClassAtom( FOD:FoodOrDrink
                                       Variable( <urn:swrl#f> ) )
                            ObjectPropertyAtom( <http://ontologies.baall.de/Hyperontology/Diet.owl#isAllowedFODSource_1discouraged>
                                                Variable( <urn:swrl#p> )
                                                Variable( <urn:swrl#bf> ) )
                            ClassAtom( <http://ontologies.baall.de/Hyperontology/DUL.owl#NaturalPerson>
                                       Variable( <urn:swrl#p> ) )
                            ObjectPropertyAtom( <http://ontologies.baall.de/Hyperontology/Object.owl#comprisesConstituent_2MediumHigh_or_greater_Intensity>
                                                Variable( <urn:swrl#f> )
                                                Variable( <urn:swrl#bf> ) ) ) Head( ObjectPropertyAtom( <http://ontologies.baall.de/Hyperontology/Diet.owl#isAllowedFOD_1discouraged>
                                                                                                        Variable( <urn:swrl#p> )
                                                                                                        Variable( <urn:swrl#f> ) ) ) )
```
The Body and the Head should be aligned.

**Comment by mcodescu** [comment]
I have created a toy ontology with SWRL rules in Protege and I saved in Manchester syntax, result below. Then I ran it through Hets via GHCI and I asked it to display the SWRL rules only. I got the following result:
```
[DLSafeRule( Body( Atom Atom Atom )
             Head( ObjectPropertyAtom( <http://www.semanticweb.org/mcodescu/ontologies/2021/9/untitled-ontology-71#hasVal>
                                       Variable( <b> )
                                       <http://www.semanticweb.org/mcodescu/ontologies/2021/9/untitled-ontology-71#v> ) ) ),
 DLSafeRule( Body( Atom Atom Atom
                   ObjectPropertyAtom( <http://www.semanticweb.org/mcodescu/ontologies/2021/9/untitled-ontology-71#hasVal>
                                       Variable( <b> )
                                       <http://www.semanticweb.org/mcodescu/ontologies/2021/9/untitled-ontology-71#v> ) )
             Head( Atom ) )]

```
Why aren't all atoms displayed correctly?

```
Prefix: owl: <http://www.w3.org/2002/07/owl#>
Prefix: rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
Prefix: rdfs: <http://www.w3.org/2000/01/rdf-schema#>
Prefix: xml: <http://www.w3.org/XML/1998/namespace>
Prefix: xsd: <http://www.w3.org/2001/XMLSchema#>



Ontology: <http://www.semanticweb.org/mcodescu/ontologies/2021/9/untitled-ontology-71>


AnnotationProperty: <http://swrl.stanford.edu/ontologies/3.3/swrla.owl#isRuleEnabled>

    
AnnotationProperty: rdfs:comment

    
AnnotationProperty: rdfs:label

    
Datatype: xsd:boolean

    
Datatype: xsd:string

    
ObjectProperty: <http://www.semanticweb.org/mcodescu/ontologies/2021/9/untitled-ontology-71#hasVal>

    Domain: 
        <http://www.semanticweb.org/mcodescu/ontologies/2021/9/untitled-ontology-71#T>
    
    Range: 
        <http://www.semanticweb.org/mcodescu/ontologies/2021/9/untitled-ontology-71#Val>
    
    
ObjectProperty: <http://www.semanticweb.org/mcodescu/ontologies/2021/9/untitled-ontology-71#p>

    Domain: 
        <http://www.semanticweb.org/mcodescu/ontologies/2021/9/untitled-ontology-71#S>
    
    Range: 
        <http://www.semanticweb.org/mcodescu/ontologies/2021/9/untitled-ontology-71#T>
    
    
ObjectProperty: <http://www.semanticweb.org/mcodescu/ontologies/2021/9/untitled-ontology-71#pv>

    SubPropertyOf: 
        <http://www.semanticweb.org/mcodescu/ontologies/2021/9/untitled-ontology-71#p>
    
    
Class: <http://www.semanticweb.org/mcodescu/ontologies/2021/9/untitled-ontology-71#S>

    
Class: <http://www.semanticweb.org/mcodescu/ontologies/2021/9/untitled-ontology-71#T>

    SubClassOf: 
        <http://www.semanticweb.org/mcodescu/ontologies/2021/9/untitled-ontology-71#hasVal> some <http://www.semanticweb.org/mcodescu/ontologies/2021/9/untitled-ontology-71#Val>
    
    
Class: <http://www.semanticweb.org/mcodescu/ontologies/2021/9/untitled-ontology-71#Val>

    
Individual: <http://www.semanticweb.org/mcodescu/ontologies/2021/9/untitled-ontology-71#v>

    Types: 
        <http://www.semanticweb.org/mcodescu/ontologies/2021/9/untitled-ontology-71#Val>
    
    
Rule: 
    <http://www.semanticweb.org/mcodescu/ontologies/2021/9/untitled-ontology-71#S>(?<A>), <http://www.semanticweb.org/mcodescu/ontologies/2021/9/untitled-ontology-71#T>(?<b>), <http://www.semanticweb.org/mcodescu/ontologies/2021/9/untitled-ontology-71#pv>(?<a>, ?<b>) -> <http://www.semanticweb.org/mcodescu/ontologies/2021/9/untitled-ontology-71#hasVal>(?<b>, <http://www.semanticweb.org/mcodescu/ontologies/2021/9/untitled-ontology-71#v>)

Rule: 
    <http://www.semanticweb.org/mcodescu/ontologies/2021/9/untitled-ontology-71#S>(?<a>), <http://www.semanticweb.org/mcodescu/ontologies/2021/9/untitled-ontology-71#T>(?<b>), <http://www.semanticweb.org/mcodescu/ontologies/2021/9/untitled-ontology-71#p>(?<a>, ?<b>), <http://www.semanticweb.org/mcodescu/ontologies/2021/9/untitled-ontology-71#hasVal>(?<b>, <http://www.semanticweb.org/mcodescu/ontologies/2021/9/untitled-ontology-71#v>) -> <http://www.semanticweb.org/mcodescu/ontologies/2021/9/untitled-ontology-71#pv>(?<a>, ?<b>)

```

File: OWL2/StaticAnalysis.hs [hunk 1]
```diff
@@ -303,28 +303,28 @@ checkDLAtom s mVars atom = case atom of
         AS.Variable v -> if Set.member i (concepts s)
             then return $ AS.ClassAtom (AS.Expression v) (AS.IVar v)
```
Comment (mcodescu) [inline OWL2/StaticAnalysis.hs:304]: Looks like the first v should be i.

Comment (daniilsvc) [reply OWL2/StaticAnalysis.hs]: Done

**Review by mcodescu — APPROVED** [review]
:+1: 
Looks good, nice done!

**Pull request merged**